asf_tools/composite.py

```diff
@@ -163,58 +164,57 @@ def make_composite(outfile, infiles=None, path=None, pol=None, resolution=None):
+        full_ul, full_lr, full_trans, full_proj = get_full_extent(raster_info)
 
-    nx = int(abs(full_ul[0] - full_lr[0]) // resolution)
-    ny = int(abs(full_ul[1] - full_lr[1]) // resolution)
+        nx = int(abs(full_ul[0] - full_lr[0]) // resolution)
+        ny = int(abs(full_ul[1] - full_lr[1]) // resolution)
 
-    outputs = np.zeros((ny, nx))
-    weights = np.zeros(outputs.shape)
-    counts = np.zeros(outputs.shape, dtype=np.int8)
+        outputs = np.zeros((ny, nx))
+        weights = np.zeros(outputs.shape)
+        counts = np.zeros(outputs.shape, dtype=np.int8)
 
-    logging.info("Calculating output values")
-    for raster, info in raster_info.items():
-        logging.info(f"Processing raster {raster}")
-        logging.info(f"Raster upper left: {info['cornerCoordinates']['upperLeft']}; "
-                     f"lower right: {info['cornerCoordinates']['lowerRight']}")
-
-        logging.info(f"Reading raster values {raster}")
-        rds = gdal.Open(raster)
-        values = rds.GetRasterBand(1).ReadAsArray()
-        del rds  # How to close w/ gdal
-
-        raster_split = raster.split('_')
-        raster_pol = raster_split[-1].split('.')[0]
-        area_suffix = raster_split[-1].replace(raster_pol, 'area')
-        area_raster = '_'.join(raster_split[:-1] + [area_suffix])
-
-        logging.info(f"Reading area raster {area_raster}")
-        ads = gdal.Open(area_raster)
-        areas = ads.GetRasterBand(1).ReadAsArray()
-        del ads  # How to close w/ gdal
-
-        ulx, uly = info['cornerCoordinates']['upperLeft']
-        y_index_start = int((full_ul[1] - uly) // resolution)
-        y_index_end = y_index_start + values.shape[0]
-
-        x_index_start = int((ulx - full_ul[0]) // resolution)
-        x_index_end = x_index_start + values.shape[1]
-
-        logging.info(
-            f"Placing values in output grid at {y_index_start}:{y_index_end} and {x_index_start}:{x_index_end}"
-        )
-
-        temp = 1.0/areas
-        temp[values == 0] = 0
-        mask = np.ones(values.shape, dtype=np.uint8)
-        mask[values == 0] = 0
-
-        outputs[y_index_start:y_index_end, x_index_start:x_index_end] += values * temp
-        weights[y_index_start:y_index_end, x_index_start:x_index_end] += temp
-        counts[y_index_start:y_index_end, x_index_start:x_index_end] += mask
+        logging.info("Calculating output values")
+        for raster, info in raster_info.items():
+            logging.info(f"Processing raster {raster}")
+            logging.info(f"Raster upper left: {info['cornerCoordinates']['upperLeft']}; "
+                         f"lower right: {info['cornerCoordinates']['lowerRight']}")
+
+            logging.info(f"Reading raster values {raster}")
+            rds = gdal.Open(raster)
+            values = rds.GetRasterBand(1).ReadAsArray()
+            del rds  # How to close w/ gdal
+
+            area_raster = '_'.join(raster.split('_')[:-1] + ['area.tif'])
```

Comment:
This business logic is duplicated in `reproject_to_target()`.  Should either be a `get_area_raster(input_raster)` function, or `reproject_to_target()` should return paths to the area rasters in it's payload